### DISCUSSION TITLE
a Gitflow-inspired restructure proposal

### DISCUSSION THREAD

**Discussion by jejellyroll-fr** [jejellyroll-fr/fpdb-3#26] [development]
Hello fellow project contributors

here is a Gitflow-inspired restructure proposal, for better collaboration

**Main branch (main) :**

Contains only stable, production-ready code
Used for releases

**Development branch (development) :**

Main branch for ongoing development
All other branches derive from it and are merged with it

**Feature branches :**

Created from the development branch
Named feature/feature-name
Merged into development once completed

**Hotfix/ branches:**

Created from main to quickly fix critical production bugs
Merged into main and development

**Bugfix/ branches:**

Created from development to fix non-critical bugs
Merged into development

**Release branches (release/) :**

Created from dev when a version is ready for release
Enables final tests and adjustments to be made before merging into main

@sf-87 @brunetton I await your feedback before implementation

**Comment by sf-87** [comment]
Seems good to me

**Comment by brunetton** [comment]
I'm very happy to read that :) I'm using these organization for more than 10 years, since my first job when I discovered this now famous [successful Git branching model](https://nvie.com/posts/a-successful-git-branching-model/) by Vincent Driessen.

So, all good to me !
This should go to the README or wiki in contribute section

**Comment by brunetton** [comment]
Hi @jejellyroll-fr 

Little doubt here: do you prefer bugfix (and all other branches names) to be uppercase ? Or lowercase ? I thought it was lower, as `development` is lower, `main` is lower by default, and `Release branches (release/) :` seems to indicate that release branches should be lower too.

Anyway, that's what I tried to make clear [in wiki page](https://github.com/jejellyroll-fr/fpdb-3/wiki/How-to-contribute:-a-Gitflow%E2%80%90inspired-struture) about Gitflow

But I just seen a king of branch name conflict in recent Git history:

![image](https://github.com/user-attachments/assets/761eb6cf-d9e3-4b2b-8f3e-adcb12c3c146)

**Reply by jejellyroll-fr** [reply]
hello Bruno

this is a mistake

the wiki is perfect

thanks
